packages/service-library/tests/conftest.py

```diff
@@ -79,12 +94,17 @@ async def _(database: RedisDatabase) -> AsyncIterator[RedisClientSDK]:
         assert client.redis_dsn == redis_resources_dns
         await client.setup()
 
-        await client.redis.flushall()
-
         yield client
 
-        # cleanup, properly close the clients
-        await client.redis.flushall()
         await client.shutdown()
 
-    return _
+    async def _cleanup_redis_data(clients_manager: RedisClientsManager) -> None:
+        for db in RedisDatabase:
+            await clients_manager.client(db).redis.flushall()
+
+    async with RedisClientsManager(
+        {RedisManagerDBConfig(db) for db in RedisDatabase}, redis_service
+    ) as clients_manager:
+        await _cleanup_redis_data(clients_manager)
+        yield _
+        await _cleanup_redis_data(clients_manager)
```

Comment:
I still don't really like that pattern.
That means that it cannot be used together with something that fills the redis database, as this will cleanup everything. The problem comes from badly designed fixtures that do not clean up after their tests...